hbase-server/src/main/java/org/apache/hadoop/hbase/coprocessor/MasterObserver.java

```diff
@@ -508,35 +508,57 @@ default void postAssign(final ObserverContext<MasterCoprocessorEnvironment> ctx,
    * Called prior to unassigning a given region.
    * @param ctx the environment to interact with the framework and master
    * @param regionInfo
-   * @param force whether to force unassignment or not
    */
   default void preUnassign(final ObserverContext<MasterCoprocessorEnvironment> ctx,
-      final RegionInfo regionInfo, final boolean force) throws IOException {}
+    final RegionInfo regionInfo) throws IOException {}
 
   /**
    * Called after the region unassignment has been requested.
    * @param ctx the environment to interact with the framework and master
    * @param regionInfo
+   */
+  default void postUnassign(final ObserverContext<MasterCoprocessorEnvironment> ctx,
+    final RegionInfo regionInfo) throws IOException {}
+
+  /**
+   * Called prior to unassigning a given region.
+   * @param ctx the environment to interact with the framework and master
+   * @param regionInfo region info
+   * @param force whether to force unassignment or not
+   * @deprecated since 2.4.0 and will be removed in 4.0.0.
+   *   Use {@link #preUnassign(ObserverContext, RegionInfo)} instead.
+   */
+  @Deprecated
+  default void preUnassign(final ObserverContext<MasterCoprocessorEnvironment> ctx,
+    final RegionInfo regionInfo, final boolean force) throws IOException {}
+
+  /**
+   * Called after the region unassignment has been requested.
+   * @param ctx the environment to interact with the framework and master
+   * @param regionInfo region info
    * @param force whether to force unassignment or not
+   * @deprecated since 2.4.0 and will be removed in 4.0.0.
```

Comment:
This class is IA.LimitedPrivate, so we could choose to break compat for 2.4.0 and just remove this method for coprocessors.

If we are keeping it, when should a coprocessor expect this specific version of `postUnassign` to get called? would it be called instead of the version without the force parameter or in addition to? These details should be incldued in the javadoc.

Reply:
Make sense, let's remove it. Thanks.